subsystem/kubeapi_test.go

```diff
@@ -644,18 +646,77 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		Eventually(func() string {
 			return getInfraEnvCRD(ctx, kubeClient, infraEnvKubeName).Status.ISODownloadURL
 		}, "15s", "5s").Should(Not(BeEmpty()))
+		// adding sleep to be sure  that the backend function GenerateClusterISOInternal() will not reject the
+		// request with "request came too soon after previous request"
+		time.Sleep(2 * bminventory.WindowBetweenRequestsInSeconds)
```

Comment:
not sure if it's needed, you are checking eventually and consistently so why do you need this wait?

Reply:
The problem here was with the infraEnv controller error handling.
It had `Requeue=False` for HTTP 409, which prevented the follow-up update.
I made additional changes and updated the commit-msg to reflect that.

Reply:
Sounds like a bug, in case we do have changes we will just skip them, the CR will show the right value but the ISO will be old. 
It looks like in case we updated something then we need to make sure that ISO is generated, even if we get 409 we need to make sure that we reconcile again.